isthmus/src/main/java/io/substrait/isthmus/expression/WindowFunctionConverter.java

```diff
@@ -0,0 +1,243 @@
+package io.substrait.isthmus.expression;
+
+import com.google.common.collect.ImmutableList;
+import io.substrait.expression.Expression;
+import io.substrait.expression.ExpressionCreator;
+import io.substrait.expression.FunctionArg;
+import io.substrait.expression.ImmutableExpression;
+import io.substrait.expression.ImmutableWindowBound;
+import io.substrait.expression.WindowBound;
+import io.substrait.expression.WindowFunctionInvocation;
+import io.substrait.function.SimpleExtension;
+import io.substrait.isthmus.SubstraitRelVisitor;
+import io.substrait.isthmus.TypeConverter;
+import io.substrait.proto.AggregateFunction;
+import io.substrait.relation.Aggregate;
+import io.substrait.type.Type;
+import java.util.Collections;
+import java.util.List;
+import java.util.Optional;
+import java.util.function.Function;
+import java.util.stream.Stream;
+import org.apache.calcite.rel.RelCollations;
+import org.apache.calcite.rel.RelFieldCollation;
+import org.apache.calcite.rel.RelNode;
+import org.apache.calcite.rel.core.AggregateCall;
+import org.apache.calcite.rel.type.RelDataType;
+import org.apache.calcite.rel.type.RelDataTypeFactory;
+import org.apache.calcite.rex.RexBuilder;
+import org.apache.calcite.rex.RexFieldCollation;
+import org.apache.calcite.rex.RexNode;
+import org.apache.calcite.rex.RexOver;
+import org.apache.calcite.rex.RexSlot;
+import org.apache.calcite.rex.RexWindowBound;
+
+public class WindowFunctionConverter
+    extends FunctionConverter<
+        SimpleExtension.WindowFunctionVariant,
+        WindowFunctionInvocation,
+        WindowFunctionConverter.WrappedAggregateCall> {
+
+  private AggregateFunctionConverter aggregateFunctionConverter;
```

Comment:
Maybe add a comment explaining that aggregate function call could appear as a Window Function, and as such we wrap an `AggregateFunctionConverter` inside a `WindowFunctionConverter`.